src/lib/zf/attr.c

```diff
@@ -281,14 +282,25 @@ int zf_attr_set_str(struct zf_attr* attr, const char* name, const char* val)
       return -ENOMSG;
     }
     char** p = get_field_str(attr, f);
-    free(*p);
     if( val != NULL ) {
+      if( strcmp(name, "log_file")==0 ) {
+        int rc = zf_log_redirect(val);
+        if( rc < 0 ) {
+          zf_log_stack_err(NO_STACK, "%s: Failed to redirect logging: %s\n",
+                          __func__, strerror(-rc));
+          /* Failure here is non-fatal. */
+          return rc;
+        }
+      }
+      free(*p);
       *p = strdup(val);
       if( *p == NULL )
```

Comment:
there is a small chance the strdup fails after changing the value of the attr holding the file name is out of sync with where it is actually going.  you would need to strdup to a temp val first and then assign *p to it later

Reply:
I opted for a slightly different solution where code check first the potential failure of zf_log_redirect and then the potential failure of strdup. In this way, it saves the app from performing the most expensive operation first (strdup) and having to eventually free the resources again if the zf_log_redirect were to fail afterwards.

Reply:
It is broken as it currently stands. If zf_log_redirect fails then we have free(*p) but it is not set to NULL.
The original point is not addressed (can redirect but strdup fails).  Are you saying this doesn't matter because you don't use local attribute name?